fix: LEAP-2009: Respect granularity when resizing NER spans

hlomzik opened the pull request:
Fix to reuse the same funtionality we use in `captureSelection()`:
- `trimSelection()` (the secret sauce!) to fix selection to point only to text nodes;
- `applyTextGranularity()` to the selection expading it if needed

We need `trimSelection()` so selection.move() will work properly.
With non-text nodes it jumps into inner/outer nodes instead of shifting symbol by symbol.

Also set cursor to "resize" for the whole document during resize.

hlomzik committed:
fix: LEAP-2009: Respect granularity when resizing NER spans

hlomzik committed:
Apply granularity only for selections inside Text tag

Otherwise it would hang the browser because of a very complex inner logic.

hlomzik committed:
Also apply `trimSelection()`, that's the secret sauce!

It fixes selection to point only to text nodes, so selection.move() will work properly.
With non-text nodes it jumps into inner/outer nodes instead of shifting symbol by symbol.

hlomzik committed:
Also set cursor to "resize" for the whole document during resize

MihajloHoma commented:
/git merge

> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14515811739)
> Successfully merged:  create mode 100644 web/libs/ui/src/utils/utils.ts

MihajloHoma committed:
Merge branch 'develop' into 'fb-leap-2009/granularity'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14515811739

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14515892252)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14515892252

MihajloHoma commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14530668488)
> Successfully merged:  4 files changed, 48 insertions(+), 19 deletions(-)

MihajloHoma committed:
Merge branch 'develop' into 'fb-leap-2009/granularity'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14530668488

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14530711986)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14530711986

hlomzik committed:
Merge branch 'develop' into fb-leap-2009/granularity

hlomzik committed:
Fix overselection with sentence granularity

Handles were messing up with ranges and selection, so we have to remove them earlier.

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14615043556)

AndrejOros committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14615043556

AndrejOros committed:
Merge branch 'develop' into 'fb-leap-2009/granularity'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14615043556

hlomzik committed:
Fix silly error preventing region from saving

hlomzik committed:
Fix linting

hlomzik committed:
Another fix for silly error

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14635186966)

MihajloHoma committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14635186966

MihajloHoma committed:
Merge branch 'develop' into 'fb-leap-2009/granularity'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14635186966

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14637483857)
> Successfully merged: Already up to date.
